src/components/Map/TileChartMap/TileChartMap.spec.js

```diff
@@ -0,0 +1,254 @@
+import { TileChartMap } from './TileChartMap';
+import React from 'react';
+import { testRender as render, screen } from '../../../testUtils/test-utils';
+import userEvent from '@testing-library/user-event';
+import { merge } from '../../../testUtils/functionHelpers';
+import { defaultMap } from '../../../reducers/map/map';
+import { defaultQuery } from '../../../reducers/query/query';
+import { defaultView } from '../../../reducers/view/view';
+import { mapResults } from './__fixtures__/mapResults';
+import { waitFor } from '@testing-library/react';
+import { GEO_NORM_PER1000 } from '../../../constants';
+import * as analyticsActions from '../../../utils';
+import * as mapActions from '../../../actions/map';
+
+describe('TileChartMap', () => {
+  const renderComponent = (newMapState, newQueryState, newViewState) => {
+    merge(newMapState, defaultMap);
+    merge(newQueryState, defaultQuery);
+    merge(newViewState, defaultView);
+
+    const data = {
+      map: newMapState,
+      query: newQueryState,
+      view: newViewState,
+    };
+    render(<TileChartMap />, {
+      preloadedState: data,
+    });
+  };
+
+  it('renders empty set without crashing', () => {
+    renderComponent();
+    expect(screen.getByTestId('tile-chart-map')).toBeInTheDocument();
+    expect(screen.getByTestId('tile-chart-map')).not.toHaveClass('print');
+    expect(
+      screen.getByText('Map of unspecified region with 1 data series.')
+    ).toBeInTheDocument();
+  });
+
+  it('renders print mode', () => {
+    renderComponent({}, {}, { isPrintMode: true });
+    expect(screen.getByTestId('tile-chart-map')).toBeInTheDocument();
+    expect(screen.getByTestId('tile-chart-map')).toHaveClass('print');
+  });
+
+  it('renders map with complaint counts', async () => {
+    const analyticsSpy = jest
+      .spyOn(analyticsActions, 'sendAnalyticsEvent')
+      .mockImplementation(() => jest.fn());
+
+    const addStateFilterSpy = jest
+      .spyOn(mapActions, 'addStateFilter')
+      .mockImplementation(() => jest.fn());
+
+    const newMap = {
+      results: {
+        state: mapResults,
+      },
+    };
+
+    const newView = {
+      isPrintMode: false,
+      width: 1000,
+    };
+
+    renderComponent(newMap, {}, newView);
+    expect(screen.getByTestId('tile-chart-map')).toBeInTheDocument();
+    expect(screen.getByTestId('tile-chart-map')).not.toHaveClass('print');
+    await waitFor(() => {
```

Comment:
I think it might be better practice to write it this way:
```
expect(await screen.findByText('FL')).toBeInTheDocument();
```
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#:~:text=Using%20waitFor%20to%20wait%20for%20elements%20that%20can%20be%20queried%20with%20find*

Reply:
I stumbled across the article recently when debugging some of my tests

Reply:
Updated.